src/core/geometry/qgssfcgalengine.h

```diff
@@ -0,0 +1,593 @@
+/***************************************************************************
+                        qgssfcgalengine.h
+  -------------------------------------------------------------------
+    begin                : May 2025
+    copyright            : (C) 2025 by Oslandia
+    email                : benoit dot de dot mezzo at oslandia dot com
+    email                : jean dot felder at oslandia dot com
+    email                : loic dot bartoletti at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifdef WITH_SFCGAL
+#ifndef QGSSFCGALENGINE_H
+#define QGSSFCGALENGINE_H
+
+#define SIP_NO_FILE
+
+#include <functional>
+#include <SFCGAL/capi/sfcgal_c.h>
+
+#include "qgspoint.h"
+#include "qgsvector3d.h"
+#include "qgis_core.h"
+#include "qgsgeometry.h"
+
+class QgsLineString;
+class QgsPolygon;
+class QgsGeometry;
+class QgsGeometryCollection;
+class QgsSfcgalGeometry;
+
+/// check if \a ptr is not null else add stacktrace entry and return the \a default_obj
+#define CHECK_NOT_NULL( ptr, default_obj )                                                                               \
+  if ( !( ptr ) )                                                                                                        \
```

Comment:
elsewhere too:

```suggestion
  if ( !ptr )                                                                                                        \
```

Reply:
I add the parenthesis due to a clang-tidy warning

Reply:
Oh, I see it's all in a macro. Can you avoid the macro here and use an inline function instead? Macros aren't ideal as they break IDE understanding of the code.

Reply:
Ok, fair enough!